lgc/patch/PatchEntryPointMutate.cpp

```diff
@@ -524,11 +524,18 @@ void PatchEntryPointMutate::fixupUserDataUses(Module &module) {
         Value *descSetVal = nullptr;
         if (inst && inst->getFunction() == &func) {
           auto call = cast<CallInst>(inst);
+          Value *highHalf = call->getArgOperand(1);
           if (descriptorSet.entryArgIdx != 0) {
-            // The descriptor set is unspilled, and uses an entry arg. Set builder to insert the 32-to-64
-            // extension code at the start of the function.
+            // The descriptor set is unspilled, and uses an entry arg.
             descSetVal = func.getArg(descriptorSet.entryArgIdx);
-            builder.SetInsertPoint(addressExtender.getFirstInsertionPt());
+            if (isa<ConstantInt>(highHalf)) {
+              // Set builder to insert the 32-to-64 extension code at the start of the function.
+              builder.SetInsertPoint(addressExtender.getFirstInsertionPt());
+            } else {
+              // Set builder to insert the 32-to-64 extension code after the instruction containing the high half.
```

Comment:
Not required for this PR, but I wonder if it would be an improvement in a future commit to move/clone the reloc call to the start of the function so the 32-to-64 extension code can still go at the start. We ought to be able to rely on some core LLVM optimization to hoist up common descriptor table extends, but I think that doesn't always work, which is why we try and insert it at the start here. Then CSE can trivially common them up.